test/ConnectionTest.cpp

```diff
@@ -0,0 +1,55 @@
+#include <cassert>
+#include <gtest/gtest.h>
+
+#include "DBConnection.hpp"
+
+namespace
+{
+    int getConnectionInfo(const IDBConnection& dbConnection)
+    {
+        return dbConnection.query();
+    }
+
+    std::string getConnectionAdvancedInfo(const IDBConnection& dbConnection)
+    {
+        assert(dynamic_cast<const MySqlDBConnection*>(&dbConnection));
```

Comment:
We need `assert(dynamic_cast<const MySqlDBConnection*>(&dbConnection))` here to prevent problems that may arise

Reply:
@Morheit I understand that. What I mean is you have to either test one cast or another, otherwise `static_cast` is meaningless in this context as you already are able to use the result of the dynamic cast

Reply:
@AGaliuzov Expected profit of this implementation is that we will detect errors in the debug build and avoid the runtime overhead in a release build

Reply:
@AGaliuzov, In this case – I tried to simulate the situation faced and resolved by @malirod. How I understood, your proposition is to add `dynamic_cast` against current realization?